src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,98 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int INITIAL_CAPACITY = 16;
+    private static final float DEFAULT_LOAD_FACTOR = 0.75f;
+    private static final int MULTIPLICATION_FACTOR = 2;
+    private Node<K, V> next;
+    private Node<K, V>[] table;
+    private int size;
+
+    public MyHashMap() {
+        table = (Node<K, V>[]) new Node[INITIAL_CAPACITY];
+    }
 
     @Override
     public void put(K key, V value) {
-
+        checkSize();
+        addNode(key, value);
+        size++;
     }
 
     @Override
     public V getValue(K key) {
+        int hash = hash(key);
+        int index = indexOfMapArray(hash);
+        Node<K, V> nodeSearch = table[index];
+        while (nodeSearch != null) {
+            if (Objects.equals(nodeSearch.key, key)) {
+                return nodeSearch.value;
+            }
+            nodeSearch = nodeSearch.next;
+        }
         return null;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    private void checkSize() {
+        if (size >= (int) (table.length * DEFAULT_LOAD_FACTOR)) {
+            Node<K, V>[] copiedTable = table;
+            table = (Node<K, V>[]) new Node[copiedTable.length * MULTIPLICATION_FACTOR];
+            for (int i = 0; i < copiedTable.length; i++) {
+                while (copiedTable[i] != null) {
+                    addNode(copiedTable[i].key, copiedTable[i].value);
+                    copiedTable[i] = copiedTable[i].next;
+                }
+            }
+        }
+    }
+
+    private void addNode(K key, V value) {
+        int hash = hash(key);
+        int index = indexOfMapArray(hash);
+        Node<K, V> node = new Node<>(hash, key, value, null);
+        if (table[index] == null) {
+            table[index] = node;
+        } else {
+            Node<K, V> nodeSearch = table[index];
+            while (nodeSearch.next != null || Objects.equals(nodeSearch.key, key)) {
+                if (Objects.equals(nodeSearch.key, key)) {
+                    size--;
+                    nodeSearch.value = value;
+                    return;
+                }
+                nodeSearch = nodeSearch.next;
+            }
+            nodeSearch.next = node;
+        }
+    }
+
+    private int indexOfMapArray(int hash) {
```

Comment:
```suggestion
    private int getBucketIndex(int hash) {
```
method names should be verbs